Lorem Ipsum XR Translator/Assets/Scripts/Util/Cache/LRUCache.cs

```diff
@@ -0,0 +1,97 @@
+        /// Inner representation of a cache entry.
+        /// </summary>
+        private class Item
+        {
+            /// <summary>
+            /// Gets or sets the key of the entry.
+            /// </summary>
+            public K Key { get; set; }
+
+            /// <summary>
+            /// Gets or sets the value of the entry.
+            /// </summary>
+            public V Value { get; set; }
+        }
+
+        // Maximum number of items that can be stored in the cache.
+        private readonly int _cap;
+
+        // Dictionary stores the key and corresponding node in the linked list for O(1) access.
+        private readonly Dictionary<K, LinkedListNode<Item>> _cache;
+
+        // LinkedList keeps track of the LRU order with most recently used items at the end.
+        private readonly LinkedList<Item> _lruList;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LRUCache{K, V}"/> class with the specified capacity.
+        /// </summary>
+        /// <param name="capacity">The maximum number of items that the cache can hold.</param>
+        public LRUCache(int capacity)
+        {
+            this._cap = capacity;
+            this._cache = new Dictionary<K, LinkedListNode<Item>>(capacity);
+            this._lruList = new LinkedList<Item>();
+        }
+
+        /// <summary>
+        /// Retrieves the value associated with the specified key from the cache.
+        /// </summary>
+        /// <param name="key">The key of the item to retrieve.</param>
+        /// <returns>The value associated with the key if the key exists; otherwise, the default value for type <typeparamref name="V"/>.</returns>
+        public V Get(K key)
+        {
+            if (this._cache.TryGetValue(key, out var node))
+            {
+                var value = node.Value.Value;
+                this._lruList.Remove(node);
+                this._lruList.AddLast(node);
+                return value;
+            }
+            return default(V);
+        }
+
+        /// <summary>
+        /// Adds or updates the value associated with the specified key in the cache.
+        /// If the cache has reached its capacity, it will remove the least recently used item before performing the addition.
+        /// </summary>
+        /// <param name="key">The key of the item to add or update.</param>
+        /// <param name="val">The value to associate with the key.</param>
+        public void Put(K key, V val)
+        {
+            if (this._cache.Count >= this._cap)
+            {
+                // Remove the least recently used item from the cache when it's full.
+                var lastNode = this._lruList.First;
+                this._cache.Remove(lastNode.Value.Key);
+                this._lruList.RemoveFirst();
+            }
+
+            var newItem = new LinkedListNode<Item>(new Item { Key = key, Value = val });
+            this._lruList.AddLast(newItem);
+            this._cache.Add(key, newItem);
```

Comment:
This doesn't handle updating already existing keys. I suggest modifying this to the following:

`

            if (Get(key) != null) // Item already exists
            {
                var newItem = new LinkedListNode<Item>(new Item { Key = key, Value = val });
                this._lruList.AddLast(newItem);
                this._cache[key] = newItem;
            }
            else // Insert new item
            {
                var newItem = new LinkedListNode<Item>(new Item { Key = key, Value = val });
                this._lruList.AddLast(newItem);
                this._cache.Add(key, newItem);
            }

Reply:
Yeah, you're right. I should handle the case where the key already exists. I looked it up and the following code creates a new pair of key-value if it doesn't exist.
```c-sharp
this._cache[key] = newItem;
```
I tested it in this playground.
https://dotnetfiddle.net/pznphR

Reply:
This looks good and is working in the shiny new unit tests.